itests/org.openhab.binding.modbus.tests/src/main/java/org/openhab/binding/modbus/tests/ModbusDataHandlerTest.java

```diff
@@ -445,7 +448,7 @@ private ModbusDataThingHandler testWriteHandlingGeneric(String start, String tra
         ModbusSlaveEndpoint endpoint = new ModbusTCPSlaveEndpoint("thisishost", 502);
 
         // Minimally mocked request
-        ModbusReadRequestBlueprint request = Mockito.mock(ModbusReadRequestBlueprint.class);
+        ModbusWriteRequestBlueprint request = Mockito.mock(ModbusWriteRequestBlueprint.class);
```

Comment:
This looks like a correct change but the mocking is now failing
```
java.lang.ClassCastException: org.mockito.codegen.ModbusWriteRequestBlueprint$MockitoMock$1796262632 cannot be cast to org.openhab.io.transport.modbus.ModbusReadRequestBlueprint
	at org.mockito.codegen.PollTask$MockitoMock$823535341.getRequest(Unknown Source)
	at org.mockito.codegen.PollTask$MockitoMock$823535341.getRequest(Unknown Source)
	at org.openhab.binding.modbus.tests.ModbusDataHandlerTest.createPollerMock(ModbusDataHandlerTest.java:145)
	at org.openhab.binding.modbus.tests.ModbusDataHandlerTest.testWriteHandlingGeneric(ModbusDataHandlerTest.java:457)
	at org.openhab.binding.modbus.tests.ModbusDataHandlerTest.testWriteRealTransformation(ModbusDataHandlerTest.java:591)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.mockito.internal.runners.DefaultInternalRunner$1$1.evaluate(DefaultInternalRunner.java:44)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.mockito.internal.runners.DefaultInternalRunner$1.run(DefaultInternalRunner.java:74)
	at org.mockito.internal.runners.DefaultInternalRunner.run(DefaultInternalRunner.java:80)
	at org.mockito.internal.runners.StrictRunner.run(StrictRunner.java:39)
	at org.mockito.junit.MockitoJUnitRunner.run(MockitoJUnitRunner.java:163)
	at junit.framework.JUnit4TestAdapter.run(JUnit4TestAdapter.java:38)
	at junit.framework.TestSuite.runTest(TestSuite.java:252)
	at junit.framework.TestSuite.run(TestSuite.java:247)
	at aQute.junit.Activator.test(Activator.java:326)
	at aQute.junit.Activator.automatic(Activator.java:220)
	at aQute.junit.Activator.run(Activator.java:176)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at aQute.launcher.Launcher.launch(Launcher.java:385)
	at aQute.launcher.Launcher.run(Launcher.java:143)
	at aQute.launcher.Launcher.main(Launcher.java:121)
	at aQute.launcher.pre.EmbeddedLauncher.executeWithRunPath(EmbeddedLauncher.java:154)
	at aQute.launcher.pre.EmbeddedLauncher.findAndExecute(EmbeddedLauncher.java:119)
	at aQute.launcher.pre.EmbeddedLauncher.main(EmbeddedLauncher.java:51)

```

Reply:
The name confused me, but now I understand that it sets up a poller to check the results of the write task. I have reset these to the original code